### PR TITLE
Fixed Index trait and new PathBuf::new()

### PR DESCRIPTION
**Pull request by Kintaro**

### CODE REVIEW

**Comment by alexcrichton** [comment]
Thanks! We'll be sure to merge this once there's a new nightly available

File: src/serialize.rs [hunk 1]
```diff
@@ -570,14 +570,18 @@ impl Decodable for path::PathBuf {
         use std::os::unix::prelude::*;
         let bytes: Vec<u8> = try!(Decodable::decode(d));
         let s: OsString = OsStringExt::from_vec(bytes);
-        Ok(path::PathBuf::new(&s))
+        let mut p = path::PathBuf::new();
+        p.push(s);
```
Comment (Kintaro) [inline src/serialize.rs]: I'm a bit unsure about this part. I hate mut. Might it be better to use FromIterator here?

Comment (alexcrichton) [inline src/serialize.rs]: Oh this can actually just be replaced with `Ok(PathBuf::from(s))`

**Comment by Kintaro** [comment]
Ah ok, didn't know about from. Changed it now.

**Pull request merged**